src/components/image/index.jsx

```diff
@@ -2,35 +2,161 @@ import PropTypes from "prop-types";
 
 const COMPONENT_CLASS_NAME = "c-image";
 
-const Image = ({ alt, className, height, loading, src, width }) => (
-	<img
-		alt={alt}
-		className={className ? `${COMPONENT_CLASS_NAME} ${className}` : `${COMPONENT_CLASS_NAME}`}
-		height={height}
-		loading={loading}
-		src={src}
-		width={width}
-	/>
-);
+const Image = ({
+	alt,
+	className,
+	loading,
+	src,
+	resizedOptions,
+	resizerURL,
+	responsiveImages,
+	sizes,
+}) => {
+	// get the default height and width of the image
+	// use aspect ratio to generate other sizes
+	const { width, height, auth } = resizedOptions;
+
+	if (!auth) {
+		// eslint-disable-next-line no-console
+		console.error("No auth token provided for resizer");
+
+		return (
+			<img
+				alt={alt}
+				className={className ? `${COMPONENT_CLASS_NAME} ${className}` : COMPONENT_CLASS_NAME}
+				src={src}
+				width={width}
+				height={height}
+			/>
+		);
+	}
+
+	// add all options except height and width
+	const stringOptionsWithoutHeightWidth = Object.keys(resizedOptions).reduce((acc, key) => {
+		if (key === "height" || key === "width") {
+			return acc;
+		}
+
+		if (acc === "") {
+			return `?${key}=${resizedOptions[key]}`;
+		}
+
+		return `${acc}&${key}=${resizedOptions[key]}`;
+	}, "");
```

Comment:
If you remove the width and height from `resizedOptions` you can do:
`new URLSearchParams(resizedOptions).toString();`

If we can't remove width/height from resizedOptions, you could destructure them out:```
const { width, height, ...resizeParams } = resizedOptions;
const urlSearchParams = new URLSearchParams(resizeParams).toString(); // ex: 'filter=filterHere&auth=abc123'

```

Reply:
updated thanks  🆒

Reply:
let me know if this is what you imagined. I liked the width split out change

Reply:
all good I presume with approval?